packages/intrepid2/unit-test/performance/StructuredIntegration/StructuredIntegrationPerformance.cpp

```diff
@@ -0,0 +1,953 @@
+    
+    // because structured integration performs transformations within integrate(), to get a fairer comparison here we include the transformation calls.
+    fstIntegrateCall->start();
+    FunctionSpaceTools::HGRADtransformGRAD(transformedGradValues, jacobianInverse, basisGradValues);
+    transformIntegrateFlopCount += double(numCellsInWorkset) * double(numFields) * double(numPoints) * double(spaceDim) * (spaceDim - 1) * 2.0; // 2: one multiply, one add per (P,D) entry in the contraction.
+    FunctionSpaceTools::multiplyMeasure(transformedWeightedGradValues, cellMeasures, transformedGradValues);
+    transformIntegrateFlopCount += double(numCellsInWorkset) * double(numFields) * double(numPoints) * double(spaceDim); // multiply each entry of transformedGradValues: one flop for each.
+        
+    auto cellStiffnessSubview = Kokkos::subview(cellStiffness, cellRange, Kokkos::ALL(), Kokkos::ALL());
+    
+    FunctionSpaceTools::integrate(cellStiffnessSubview, transformedGradValues, transformedWeightedGradValues);
+    ExecSpaceType().fence();
+    fstIntegrateCall->stop();
+    
+    transformIntegrateFlopCount += double(numCellsInWorkset) * double(numFields) * double(numFields) * double(numPoints * spaceDim * 2 - 1); // 2: one multiply, one add per (P,D) entry in the contraction.
+    
+    cellOffset += worksetSize;
+  }
+//  std::cout << "standard integration, approximateFlopCount: " << approximateFlopCount << std::endl;
+  return cellStiffness;
+}
+
+//! returns an estimated count of the floating point operations performed.
+template<class Scalar, class PointScalar, int spaceDim, typename ExecSpaceType>
+void performStructuredQuadratureHypercubeGRADGRAD(CellGeometry<PointScalar, spaceDim, ExecSpaceType> &geometry, const int &polyOrder, const int &worksetSize,
+                                                  double &transformIntegrateFlopCount, double &jacobianCellMeasureFlopCount)
+{
+  int numVertices = 1;
+  for (int d=0; d<spaceDim; d++)
+  {
+    numVertices *= 2;
+  }
+  
+  auto initialSetupTimer = Teuchos::TimeMonitor::getNewTimer("Initial Setup");
+  initialSetupTimer->start();
+  using namespace std;
+  using FunctionSpaceTools = Intrepid2::FunctionSpaceTools<ExecSpaceType>;
+  using IntegrationTools   = Intrepid2::IntegrationTools<ExecSpaceType>;
+  // dimensions of the returned view are (C,F,F)
+  auto fs = Intrepid2::FUNCTION_SPACE_HGRAD;
+  
+  shards::CellTopology cellTopo = geometry.cellTopology();
+  
+  auto basis = Intrepid2::getBasis< Intrepid2::DerivedNodalBasisFamily<> >(cellTopo, fs, polyOrder);
+  
+  int numFields = basis->getCardinality();
+  int numCells = geometry.numCells();
+    
+  // local stiffness matrix:
+  ScalarView<Scalar,ExecSpaceType> cellStiffness("cell stiffness matrices",numCells,numFields,numFields);
+  
+  auto cubature = Intrepid2::DefaultCubatureFactory::create<ExecSpaceType>(cellTopo,polyOrder*2);
+  auto tensorCubatureWeights = cubature->allocateTensorCubatureWeights();
```

Comment:
In my understanding this works also for non-tensor quadrature, without any real added cost (?), so it would be nice to make it the new way to get quadratures. Maybe we could  rename `allocateTensoreCubatureXXX` to `allocateCubatureXXX` and then `getTensorCubature` to `getCubature`, which would overload the old version.

Reply:
Ah, I see that you already had that idea before I brought it up yesterday.  I'd overlooked this when writing my other responses yesterday.  In any case, I've now made this change.